json/src/main/java/org/jdbi/v3/json/internal/JsonArgumentFactory.java

```diff
@@ -41,7 +41,8 @@
         if (String.class.equals(type)) {
             return Optional.empty();
         }
-        String json = value == null ? null : config.get(JsonConfig.class).getJsonMapper().toJson(type, value, config);
+        String nullableJson = value == null ? null : config.get(JsonConfig.class).getJsonMapper().toJson(type, value, config);
+        String json = "null".equalsIgnoreCase(nullableJson) ? null : nullableJson; // json null -> sql null
```

Comment:
Only lowercase `null` is valid JSON here--`equalsIgnoreCase` is probably overkill